### DISCUSSION TITLE
Is Possible Running Nuclei in website authenticate?

### DISCUSSION THREAD

**Discussion by Marcio861** [projectdiscovery/nuclei#2149] [Q&A] [answered]
Is Possible Running Nuclei in website authenticate?

With owasp zap or burp suite?

**Comment by forgedhallpass** [comment, marked as answer]
Depending on the authentication scheme, you can pass in a valid authorization/session etc. header (e.g. see https://github.com/projectdiscovery/nuclei/discussions/1608).

If you are creating your own template, you can add a "manual" authentication step, and use an extractor to retrieve the value to be reused, or if the target application uses cookies, you could probably also create a dedicated template only for authenticating to a specific host and make it part of a workflow. This should also work since nuclei stores all the cookies in a cookie jar.
